framework/src/outputs/Output.C

```diff
@@ -272,14 +309,39 @@ Output::onInterval()
   if (_sync_times.find(_time) != _sync_times.end())
     output = true;
 
-  // check if enough time has passed between outputs
-  if (_time > _last_output_time && _last_output_time + _minimum_time_interval > _time + _t_tol)
-    return false;
+  // check if enough simulation time has passed between outputs
+  if (_time > _last_output_simulation_time &&
+      _last_output_simulation_time + _min_simulation_time_interval > _time + _t_tol)
+    output = false;
+
+  // check if enough wall time has passed between outputs
+  const auto now = std::chrono::steady_clock::now();
```

Comment:
I think that we need to synchronize these times. Or let rank 0 decide and then communicate its choice to all procs. If we hit this at different times on different processors, we could take different branches in this object and it would really screw things up

Reply:
Nice catch

Reply:
Shouldn't we do min?

Reply:
The purpose of the wall time checkpoint feature can be oversimplified to the following statement: "Output a checkpoint every 10 minutes of wall time a simulation runs."

On line 323 below, each processor computes the wall time since the last output was written. One processor will have the smallest wall time since last output and one will have the largest wall time since last output. I could see it working both ways (using min or max), but I'd argue that using max is the way to go since with using min, it could be longer than 10 minutes before the next output. In practice, it probably doesn't matter too much, so I'm willing to change it to min if you have an argument for it.